pkg/polling.go

```diff
@@ -112,6 +115,7 @@ type PollingObserver struct {
 	keys     KeyProvider               // provides keys to this block observer
 	coord    KeyStatusCoordinator      // key status coordinator tracks in-flight status
 	shuffler Shuffler[types.UpkeepKey] // provides shuffling logic for upkeep keys
+	filterer keepers.Coordinator       // provides filtering logic for upkeep keys
```

Comment:
We might even be able to simplify this interface since an `Observer` only interacts with `IsPending` anyway. While a `Coordinator` is a larger struct that links some things together, we don't need to define the whole component as a dependency in an `Observer`.